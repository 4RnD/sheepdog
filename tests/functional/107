#!/bin/bash

# Test dog node recovery set-throttle

. ./common

function oneTimeSetUp {
	_start_sheep 0
	_wait_for_sheep 1
	_cluster_format -c 1
}

function setUp {
	dog node recovery set-throttle -- 0 0
}

function testSetThrottleSuccess {
	setUp
	dog node recovery set-throttle -- "$@"
	[ $? -eq 0 ] || exit 1
	dog node recovery get-throttle
}

function testSetThrottleFailure {
	setUp
	! dog node recovery set-throttle -- "$@"
	[ $? -eq 0 ] || exit 1
	dog node recovery get-throttle
}

oneTimeSetUp

testSetThrottleFailure  0  1
testSetThrottleFailure  0 -1
testSetThrottleFailure  0 -2

testSetThrottleFailure  1  0
testSetThrottleFailure  1 -1
testSetThrottleFailure  1 -2

testSetThrottleFailure -1  0
testSetThrottleFailure -1  1
testSetThrottleFailure -1 -1
testSetThrottleFailure -1 -2

testSetThrottleFailure -2  0
testSetThrottleFailure -2  1
testSetThrottleFailure -2 -1
testSetThrottleFailure -2 -2

testSetThrottleFailure 4294967295 9223372036854775808
testSetThrottleFailure 4294967296 9223372036854775807
testSetThrottleFailure 4294967296 9223372036854775808

testSetThrottleSuccess 0 0
testSetThrottleSuccess 1 1
testSetThrottleSuccess 4294967295 9223372036854775807
