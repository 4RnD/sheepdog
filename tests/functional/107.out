QA output created by 107
using backend plain store
Invalid interval max (0), interval (1)
max (0), interval (0)
Invalid interval (-1)
max (0), interval (0)
Invalid interval (-2)
max (0), interval (0)
Invalid interval max (1), interval (0)
max (0), interval (0)
Invalid interval (-1)
max (0), interval (0)
Invalid interval (-2)
max (0), interval (0)
Invalid max (-1)
max (0), interval (0)
Invalid max (-1)
max (0), interval (0)
Invalid max (-1)
max (0), interval (0)
Invalid max (-1)
max (0), interval (0)
Invalid max (-2)
max (0), interval (0)
Invalid max (-2)
max (0), interval (0)
Invalid max (-2)
max (0), interval (0)
Invalid max (-2)
max (0), interval (0)
Invalid interval (9223372036854775808)
max (0), interval (0)
Invalid max (4294967296)
max (0), interval (0)
Invalid max (4294967296)
max (0), interval (0)
max (0), interval (0)
max (1), interval (1)
max (4294967295), interval (9223372036854775807)
